contents/node/nodejs.md

```diff
@@ -0,0 +1,226 @@
+# Introduction to Node.js
+
+Author: Rachael Sim
+
+# Overview
+
+*Node.js* is a JavaScript runtime built on Chrome’s V8 JavaScript engine. Node.js uses an event-driven, non-blocking I/O model that makes it lightweight and efficient.
+
+### Event-driven
+
+When an event happens e.g. an I/O event is complete, the event handler/callback function will be added to a queue where it will be subsequently scheduled to run by the [event loop](https://medium.com/the-node-js-collection/what-you-should-know-to-really-understand-the-node-js-event-loop-and-its-metrics-c4907b19da4c).
+
+### Non-Blocking
+
+When file and other I/O operations requests are made in Python or Java, they are blocking - subsequent requests (in the same thread) have to be enqueued and can only be processed after the current operation completes. In contrast, Node.js has an event loop which delegates the actual tasks to other systems (e.g. file system and databases) and it is almost never blocked after delegation. While an I/O operation is incomplete, the event loop can still process subsequent requests.
+
+### Benefits
+
+* Node.js is fast for I/O intensive tasks due to the non-blocking model.
+* As Node is single-threaded, we can avoid thread overheads and synchronization problems including deadlocks and race conditions.
+* Good support - largest ecosystem of open source libraries in the world
+* Same language for front and back-end development
+
+To find out more about when you should or should not use Node.js, checkout these articles by [netguru](https://www.netguru.co/blog/use-node-js-backend) and [Node.js foundation](https://medium.com/the-node-js-collection/why-the-hell-would-you-use-node-js-4b053b94ab8e).
+
+# Getting Started
+
+To install Node, download the installer from the [official Node.js website](https://nodejs.org/en/download/) based on your OS and run it. Complete the installation process by restarting your system.
+ ```
+ $ node -v
+ $ npm -v
+ ```
+Run the above commands in the command prompt. If the installation of Node and npm were successful, it should return the version number of the currently active Node binary (e.g. v8.9.4).
+
+# Node Package Manager (npm)
+
+Node Package Manager (npm) is used to
+* Search for node.js packages online
+* Install node.js packages from the command line, manage versions and dependencies effectively and easily
+
+To load dependencies, you will just need to run the following command in the command shell.
+```
+$ npm install
+```
+
+This command will locate the `package.json` file (a file that contains all metadata information about a Node application) in the root directory and install the dependencies specified in that file.
```

Comment:
Contents in this section up to this point are fine. Although you are giving instructions, they can serve as a general description of how easy it is to load dependencies.